app/components/layout/sidebar/sidebar.tsx

```diff
@@ -11,12 +11,56 @@ import SidebarBottom from "./bottom";
 import MenuButton from "./menu-button";
 import MenuItems from "./menu-items";
 import Overlay from "./overlay";
+import { useMediaStream } from "~/components/scan-qr/media_stream_provider";
+import QRCodeScanner from "~/components/scan-qr/QRCodeScanner";
 
 export default function Sidebar() {
   const { user, minimizedSidebar } = useLoaderData<typeof loader>();
   const [isMobileNavOpen, toggleMobileNav] = useAtom(toggleMobileNavAtom);
   const mainNavigationRef = useRef<HTMLElement>(null);
 
+  const [showScanner, setShowScanner] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [scannedSuccessfully, setScannedSuccessfully] = useState(false);
+
+  const { startMediaStream, stopMediaStream } = useMediaStream();
+
+  const isValidQRCode = (data: string) => {
```

Comment:
Can you explain this? What's this regex validating?

Reply:
to check if qr code URLs are valid, I had designed it to work with Shelf.nu URLs, local hots URLs, and any other self-hosted instances's URLs -- basically it verifies that scanned QR has valid asset URLs

Reply:
Okey. Can you abstract it into a function with a very clear name, so we know what it does. Like `checkQrLinkValidity` or sth like this. 
Also on successful scan we would need to navigate to that url where we do some server side validations, but good to know beforehand if its even valid. Just to confirm, the correct structure is: `/qr/${qrId}`